akka-http-core/src/main/scala/akka/http/impl/engine/client/HttpsProxyGraphStage.scala

```diff
@@ -48,7 +49,7 @@ private final class HttpsProxyGraphStage(targetHostName: String, targetPort: Int
 
   override def shape: BidiShape[ByteString, ByteString, ByteString, ByteString] = BidiShape.apply(sslIn, bytesOut, bytesIn, sslOut)
 
-  private val connectMsg = ByteString(s"CONNECT $targetHostName:$targetPort HTTP/1.1\r\nHost: $targetHostName\r\n\r\n")
+  private val connectMsg = ByteString(s"CONNECT $targetHostName:$targetPort HTTP/1.1\r\nHost: $targetHostName\r\n${proxyAuth.fold("")(httpCredentials ⇒ s"${`Proxy-Authorization`(httpCredentials)}\r\n")}\r\n")
```

Comment:
I wonder if we could use the internal rendering capabilities instead of the string... But can be as is for now and we'll do a separate ticket for that as an improvement perhaps